necromorphs/code/modules/weapons/guns/divet.dm

```diff
@@ -1,50 +1,235 @@
+/**
+Divet pistols
+*/
+
 /obj/item/gun/ballistic/automatic/pistol/divet
 	name = "divet pistol"
 	desc = "A Winchester Arms NK-series pistol capable of fully automatic fire."
-	icon = 'necromorphs/icons/obj/weapons/gun.dmi'
+	icon = 'necromorphs/icons/obj/weapons/ds13guns.dmi'
 	icon_state = "divet"
 	lefthand_file = 'necromorphs/icons/mob/onmob/items/lefthand_guns.dmi'
 	righthand_file = 'necromorphs/icons/mob/onmob/items/righthand_guns.dmi'
-	inhand_icon_state = "divet"
 	w_class = WEIGHT_CLASS_NORMAL
 	mag_type = /obj/item/ammo_box/magazine/divet
-	can_suppress = FALSE
+	can_suppress = TRUE
 	actions_types = list(/datum/action/item_action/toggle_firemode)
+	empty_indicator = TRUE
 	slot_flags = ITEM_SLOT_BELT|ITEM_SLOT_POCKETS
 	burst_size = 3
-	fire_delay = 3
-	show_bolt_icon = FALSE
-	fire_sound = 'necromorphs/sound/weapons/guns/divet_fire.ogg'
-	load_sound = 'necromorphs/sound/weapons/guns/divet_magin.ogg'
-	eject_sound = 'necromorphs/sound/weapons/guns/divet_magout.ogg'
+	fire_delay = 4
+	dual_wield_spread = 6
+	firing_burst = TRUE
+	fire_sound= 'sound/ds13/divet_fire.ogg'
+	load_sound = 'necromorphs/sound/weapons/guns/divet_magin.ogg'  //Old noises from 1.0, but weren't actually working and used before.
+	eject_sound = 'necromorphs/sound/weapons/guns/divet_magout.ogg' //They just sound like clockwork cult sounds.
+	suppressed_sound = 'sound/weapons/gun/general/heavy_shot_suppressed.ogg'
+
+/obj/item/gun/ballistic/automatic/pistol/divet/extended
+	mag_type = /obj/item/ammo_box/magazine/divet/extended //Figure out how to keep compatability with normal mags, but use special mag. Spcial amg types may work if this doesn't
+
+/obj/item/gun/ballistic/automatic/pistol/divet/extended
+	mag_type = /obj/item/ammo_box/magazine/divet/extended/expanded
```

Comment:
Duplicate code

Reply:
Whoops. Meant for each magtype. 
I'll take a moment to get the mags figured out